core/src/net.rs

```diff
@@ -0,0 +1,46 @@
+//! Module for netlink operations
+
+use std::process::Command;
+
+use anyhow::bail;
+use nix::unistd::Pid;
+use regex::Regex;
+
+/// Moves interface iname to the network namespace of PID
+///
+/// Keep in mind, that the interface **is not** being moved
+/// back to the original namespace. Beside this, most configuration
+/// done for the interface, including it's IP address, get lost
+/// when moving it to a different namespace.
+pub fn move_to_ns(iname: &str, pid: Pid) -> anyhow::Result<()> {
+    let cmd = Command::new("sh")
+        .arg("-c")
+        .arg("ip link set dev ".to_string() + iname + " netns " + &pid.to_string())
```

Comment:
Why call sh here, instead of `ip` with the proper `.args()`? Using `iname` this way makes this more prone to command injection vulnerabilities.

Reply:
Done. I've also added an appropriate check in 82f74549f97500954eda320d30c5bd2aeffc4642.